### PR TITLE
10_fibonacci: handling of string zero & update of tests

### PR DESCRIPTION
**Pull request by jamienorthman**
## Because
The test case for string zero ('0') was missing, and the solution, using a 'deeply equals' check for 0, returned 1 instead of 0. Also, the tests for regular number 0 for the exercise was missing yet it was in the solution directory, which made it inconsistent.

## This PR

- Fixed handling of string 0 in solution
- Added testing of string 0 to exercise and solution
- Added testing of number 0 to exercise

## Issue
Related to #356

## Additional Information
This issue was also mentioned in #356. However, both PRs address slightly different topics

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `01_helloWorld: Update test cases`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If this PR includes any changes that affect the solution of an exercise, I've also updated the solution in the `/solutions` folder

### CODE REVIEW

**Review by KelvinMvungi — COMMENTED** [review]
This works!

File: 10_fibonacci/fibonacci.spec.js [hunk 1]
```diff
@@ -16,9 +16,15 @@ describe('fibonacci', () => {
   test.skip('25th fibonacci number is 75025', () => {
     expect(fibonacci(25)).toBe(75025);
   });
+  test.skip('0th fibonacci number is o', () => {
```
Comment (fortypercenttitanium) [inline 10_fibonacci/fibonacci.spec.js:19]:
```suggestion
  test.skip('0th fibonacci number is 0', () => {
```

File: 10_fibonacci/solution/fibonacci-solution.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 const fibonacci = function(count) {
     if (count < 0) return "OOPS";
-    if (count === 0) return 0;
+    if (count == 0) return 0;
```
Comment (fortypercenttitanium) [inline 10_fibonacci/solution/fibonacci-solution.js:3]: I don't like this, we should be explicit. Let's convert count to a number first.

File: 10_fibonacci/solution/fibonacci-solution.spec.js [hunk 3]
```diff
@@ -19,16 +19,19 @@ describe('fibonacci', () => {
   test('0th fibonacci number is o', () => {
```
Comment (fortypercenttitanium) [inline 10_fibonacci/solution/fibonacci-solution.spec.js:19]:
```suggestion
  test('0th fibonacci number is 0', () => {
```
for consistency

**Review by fortypercenttitanium — CHANGES_REQUESTED** [review]
Just a couple changes. Thanks.

File: 10_fibonacci/solution/fibonacci-solution.js (hunk 2 above)
Comment (jamienorthman) [reply 10_fibonacci/solution/fibonacci-solution.js:3]: Thanks, adapting it now.

**Review by CouchofTomato — APPROVED** [review]

**Review by fortypercenttitanium — DISMISSED** [review]

**Comment by CouchofTomato** [comment]
@fortypercenttitanium Are you happy with this? I'm not versed enough with it to know if this is the intended outcome.

**Comment by fortypercenttitanium** [comment]
> @fortypercenttitanium Are you happy with this? I'm not versed enough with it to know if this is the intended outcome.

Yes it's good for me

**Review by CouchofTomato — APPROVED** [review]

**Pull request merged**